### PR TITLE
- Made encode and decode free of proof obligations so that one can do…

### PR DESCRIPTION
**Pull request by ric-almeida**
… Compute of them

- Added an exame at the bottom of Morello.v (to be deleted afterwards, it's just for illustration purposes now) showing how decode(encode(c)) != c for one c.

### CODE REVIEW

File: coq/Morello.v [hunk 1]
```diff
@@ -854,4 +850,39 @@ Module MorelloCapability <:
...
+      |}
+  |}.
+
+  Compute cap_ex_c2.
+  Definition cap_enc := encode true cap_ex_c2.
+  Compute cap_enc.
+  Definition cap_enc_dec := 
+    decode (match cap_enc with Some (l,b) => l | None => [] end) true.
+  Compute cap_enc_dec.
+
```
Comment (vzaliva) [inline coq/Morello.v:887]: @ric-almeida this part seems to be a spillover from another PR.

File: coq/Morello.v [hunk 2]
```diff
@@ -581,20 +581,17 @@ Module MorelloCapability <:
...
       end.
 
-  Program Definition decode (bytes : list ascii) (tag : bool) : option t :=
+  Definition decode (bytes : list ascii) (tag : bool) : option t :=
     if Nat.eqb (List.length bytes) 16%nat then
       let bytes := List.rev bytes in
       let bits := tag::(bool_bits_of_bytes bytes) in
       let bitsu := List.map bitU_of_bool bits in
       let w := vec_of_bits bitsu in
+      let w:mword 129 := mword_of_int (int_of_mword false w) in
```
Comment (vzaliva) [inline coq/Morello.v:590]: This may solve the obligation but introduces another re-encoding roundtrip which will affect performance.

File: coq/Morello.v [hunk 3]
```diff
@@ -673,12 +670,11 @@ Module MorelloCapability <:
     let perms :=
       List.map bitU_of_bool (MorelloPermission.to_list (cap_get_perms c)) in
     let perms := vec_of_bits perms in
+    let perms : mword MorelloPermission.perms_zlen :=
+      mword_of_int (int_of_mword false perms) in
```
Comment (vzaliva) [inline coq/Morello.v:674]: this adds another roundtrip encoding/decoding of bits to word which will affect performance.

File: coq/Morello.v [hunk 4]
```diff
@@ -760,7 +756,7 @@ Module MorelloCapability <:
     | _ => None
     end.
 
-  Program Definition encode (isexact : bool) (c : t) : option ((list ascii) * bool) :=
+  Definition encode (isexact : bool) (c : t) : option ((list ascii) * bool) :=
```
Comment (vzaliva) [inline coq/Morello.v:759]: Ive manually merger this part. Thanks.

File: coq/Morello.v (hunk 2 above)
Comment (ric-almeida) [reply coq/Morello.v:590]: Same comment I made below.

File: coq/Morello.v (hunk 1 above)
Comment (ric-almeida) [reply coq/Morello.v:887]: Yes, is it possible to reject just this part here from the interface?
